components/widgets/public-private-selector/public-private-selector.tsx

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```

Comment:
Based on https://github.com/mattermost/mattermost-plugin-playbooks/blob/master/webapp/src/components/backstage/public_private_selector.tsx from Playbooks, handling use cases for both projects.